core/src/main/java/io/kestra/core/runners/pebble/functions/FileExistsFunction.java

```diff
@@ -0,0 +1,99 @@
+package io.kestra.core.runners.pebble.functions;
+
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.utils.Slugify;
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Function;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+@Singleton
+public class FileExistsFunction implements Function {
+    private static final String ERROR_MESSAGE = "The 'fileSize' function expects an argument 'path' that is a path to the internal storage URI.";
```

Comment:
```suggestion
    private static final String ERROR_MESSAGE = "The 'fileExists' function expects an argument 'path' that is a path to the internal storage URI.";
```

Reply:
empty has a different meaning than blank.
and an empty file is usually a file with nothing in it, so reading the first byte and checking if it's >0 should be enought